src/script.js

```diff
@@ -22,19 +22,42 @@
       return ((v2Rect.width * v2Rect.height) - (v1Rect.width * v1Rect.height));
     });
 
-  if (videos.length === 0)
+  if (videos.length === 0) {
     return;
+  }
+
+  return videos[0];
+}
 
-  const video = videos[0];
+async function requestPictureInPicture(video) {
+  await video.requestPictureInPicture();
+  video.setAttribute('__pip__', true);
+  video.addEventListener('leavepictureinpicture', event => {
+    video.removeAttribute('__pip__');
+  }, { once: true });
+}
+
+async function maybeUpdatePictureInPictureVideo() {
+  if (!document.querySelector('[__pip__]')) {
+    return;
+  }
+  const video = findLargestPlayingVideo();
+  if (video && !video.hasAttribute('__pip__')) {
+    await requestPictureInPicture(video);
```

Comment:
Shouldn't this be:
```
  await requestPictureInPicture(video);
  resizeObserver.unobserve(current_pip_video);
  resizeObserver.observe(video);
```

It implies that we save the value returned by the `querySelector()` call above.

Reply:
How would you save the value of a video DOM element in the context of an extension script?

Reply:
The video would be using `document.querySelector('[__pip__]')` but I guess your point applies for the `ResizeObserver`. I think we could use the fact that the `ResizeObserverCallback` includes the `ResizeObserver` as a second argument and we could `unobserver()` `document.querySelector('[__pip__]]')` and instead observer the new video. This way we avoid observing all possible videos but only the current main video. WDYT? I may try this later today...

Reply:
I didn't realize the second argument of ResizeObserverCallback was observer. Thank you really much! See last commit: https://github.com/GoogleChromeLabs/picture-in-picture-chrome-extension/pull/42/commits/34970f6a40a5c516b7515736d7cc7357e0c8553a